silver/tests/spec/test_payments.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2017 Presslabs SRL
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+from collections import OrderedDict
+from decimal import Decimal
+
+from mock import patch
+
+from rest_framework import status
+from rest_framework.reverse import reverse as _reverse
+from rest_framework.test import APITestCase
+
+from django.utils import timezone
+from django.template.loader import render_to_string
+
+from silver.models import Transaction
+from silver.models.payment_processors.base import PaymentProcessorBase
+from silver.models.payment_processors.mixins import TriggeredProcessorMixin
+from silver.utils.payments import get_payment_url
+
+from silver.tests.factories import (AdminUserFactory, TransactionFactory,
+                                    PaymentMethodFactory, InvoiceFactory,
+                                    ProformaFactory, CustomerFactory)
+from silver.tests.utils import register_processor
+from silver.utils.payments import get_payment_url
+
+
+class TestPaymentUrls(APITestCase):
+    def setUp(self):
+        self.user = AdminUserFactory.create()
+        self.client.force_authenticate(user=self.user)
+
+    def test_pay_transaction_view_expired(self):
+        customer = CustomerFactory.create()
+        payment_method = PaymentMethodFactory.create(customer=customer)
+        transaction = TransactionFactory.create(payment_method=payment_method)
+
+        with patch('silver.utils.payments.datetime') as mocked_datetime:
+            mocked_datetime.utcnow.return_value = datetime.utcnow() - timedelta(days=365)
```

Comment:
Maybe leave a comment, explaining the magic: this will create an expired token such that the view will return 404. Also renaming the test to `test_pay_transaction_view_expired_token` might offer more insight about what this test does.